pandas/core/frame.py

```diff
@@ -8417,6 +8417,132 @@ def isin(self, values):
+        ----------
+        normalize : boolean, default False
+            If True then the Series returned will contain the relative
+            frequencies of the unique values.
+        sort : boolean, default True
+            Sort by frequencies.
+        ascending : boolean, default False
+            Sort in ascending order.
+        bins : integer, optional
+            Rather than count values, group them into half-open bins,
+            a convenience for ``pd.cut``, only works with single-column numeric
+            data.
+        dropna : boolean, default True
+            Don't include counts of rows containing NaN.
+
+        Returns
+        -------
+        counts : Series
+
+        See Also
+        --------
+        Series.value_counts: Equivalent method on Series.
+
+        Examples
+        --------
+
+        >>> df = pd.DataFrame({'num_legs': [2, 4, 4, 6],
+        ...                    'num_wings': [2, 0, 0, 0]},
+        ...                   index=['falcon', 'dog', 'cat', 'ant'])
+        >>> df
+                num_legs  num_wings
+        falcon         2          2
+        dog            4          0
+        cat            4          0
+        ant            6          0
+
+        >>> df.value_counts()
+        num_legs  num_wings
+        4         0            2
+        6         0            1
+        2         2            1
+        dtype: int64
+
+        >>> df.value_counts(sort=False)
+        num_legs  num_wings
+        2         2            1
+        4         0            2
+        6         0            1
+        dtype: int64
+
+        >>> df.value_counts(ascending=True)
+        num_legs  num_wings
+        2         2            1
+        6         0            1
+        4         0            2
+        dtype: int64
+
+        >>> df.value_counts(normalize=True)
+        num_legs  num_wings
+        4         0            0.50
+        6         0            0.25
+        2         2            0.25
+        dtype: float64
+
+        >>> single_col_df = df[['num_legs']]
+        >>> single_col_df.value_counts(bins=4)
+        num_legs
+        (3.0, 4.0]      2
+        (5.0, 6.0]      1
+        (1.995, 3.0]    1
+        (4.0, 5.0]      0
+        dtype: int64
+        """
+
+        # Delegate to Series.value_counts for single-column data frames.
+        if len(self.columns) == 1:
+            series = self[self.columns[0]].value_counts(
+                normalize=normalize,
+                sort=sort,
+                ascending=ascending,
+                bins=bins,
+                dropna=dropna,
+            )
+            # Move series name into its index, as happens in multi-column case.
+            return Series(
+                data=series.values,
+                index=MultiIndex.from_arrays(
+                    [series.index.values], names=[series.name]
+                ),
+            )
+
+        # Some features are only supported for single-column data.
```

Comment:
I think these checks should be done first.

I'd rather have

```python
df = pd.DataFrame({"A": [1, 2], "B": [3, 4]})
df.value_counts(dropna=False)
df[['A']].value_counts(dropna=False)
```

both raise. That way, we don't have the behavior depending on the shape.

Reply:
Done. I had to keep the special casing of single columns since otherwise you get an index that's not a multiindex.